### PR TITLE
Cb-1847 decrement count deleted messages

### PR DESCRIPTION
**Pull request by danielr18**

### CODE REVIEW

**Comment by danielr18** [comment]
Need to create migration

File: functions/src/commons/business/decrementFeedObjectCountForUsersOnDelete.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        if (currentLastSeenObject) {
+          if (
+            dayjs(deletedDiscussionMessage.createdAt.toDate()).isAfter(
+              currentLastSeenObject.createdAt.toDate()
+            )
+          ) {
+            await commonFeedObjectUserUniqueDb.update(
+              {
+                count: feedObjectUserUnique.count - 1,
```
Comment (ariel-dao) [inline functions/src/commons/business/decrementFeedObjectCountForUsersOnDelete.ts]: make sure that count can't go below 0 (shouldn't happen but we can never know for sure)

File: functions/src/commons/business/decrementFeedObjectCountForUsersOnDelete.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+              },
+              commonFeedObject.id,
+              member.userId,
+              discussion.commonId
+            );
+
+            if (feedItemFollow) {
+              const follow = await feedItemFollowsDb.update(
+                {
+                  count: feedItemFollow.count - 1,
```
Comment (ariel-dao) [inline functions/src/commons/business/decrementFeedObjectCountForUsersOnDelete.ts]: count min 0

File: functions/src/commons/db/CommonFeedObjectUserUnique/add.ts [hunk 3]
```diff
@@ -15,6 +15,15 @@ export const add = async (
 ) => {
   if (!commonId) return undefined;
 
+  const existingObjects =
+    (
+      await CommonFeedObjectUserUniqueSubcollection(commonId)
+        .where("feedObjectId", "==", commonFeedObjectUserUnique.feedObjectId)
```
Comment (ariel-dao) [inline functions/src/commons/db/CommonFeedObjectUserUnique/add.ts:21]: might need to index this

**Review by ariel-dao — CHANGES_REQUESTED** [review]
Looks good merge affter changes

File: functions/src/commons/db/CommonFeedObjectUserUnique/add.ts (hunk 3 above)
Comment (danielr18) [reply functions/src/commons/db/CommonFeedObjectUserUnique/add.ts:21]: It already exists

**Pull request merged**